### PR TITLE
update osw converter for noaa l2 product

### LINKED ISSUE #1553: update OSW converter for NOAA level-2 product from CDP
## Description

The NOAA CDP pilot has provided a Level-2 OSW product for GNSS-R wind speed.

## Requirements

backward compatibility for all current OSW products as well as capability to decode and create IODA file for this new Level-2 product

## Acceptance Criteria (Definition of Done)

IODA file is created

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

Provide a detailed description of what this PR does. What problem does it fix? What new capability does it add?

## Issue(s) addressed

Resolves #1553 

## Impact

converter augmented to decode another wind product

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

File: src/hdf5/osw_2ioda.py [hunk 1]
```diff
@@ -213,10 +213,15 @@ def get_data_source(afile):
         return 'Muon'
     elif 'constellation' in afile.attrs.keys() and "spire" in afile.attrs['constellation'].decode('UTF-8'):
         return 'Spire'
+    elif 'title' in afile.attrs.keys() and 'SPIRE' in afile.attrs['title'].decode('UTF-8'):
+        return 'SPIRE'
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
        return 'Spire-L2'
```

File: src/hdf5/osw_2ioda.py [hunk 2]
```diff
@@ -213,10 +213,15 @@ def get_data_source(afile):
...
+    else:
+        # throw error
+        pass
 
 
 def get_reference_time(afile, osw_source):
-    if osw_source in ('CYGNSS'):
+    if osw_source in ('CYGNSS') or 'SPIRE' in osw_source:
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
    if osw_source in ('CYGNSS') or 'Spire-L2' in osw_source:
```

File: src/hdf5/osw_2ioda.py [hunk 3]
```diff
@@ -235,7 +240,7 @@ def get_reference_time(afile, osw_source):
 def get_data_from_file(afile, col_names, osw_source, file_name):
 
     # Pull each data type (variable) and create a list
-    if osw_source == 'CYGNSS':
+    if osw_source == 'CYGNSS' or osw_source == 'SPIRE':
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
    if osw_source == 'CYGNSS' or osw_source == 'Spire-L2':
```

File: src/hdf5/osw_2ioda.py [hunk 4]
```diff
@@ -298,7 +303,7 @@ def adjust_dateTime(obs_DF, dat_ref):
 
 
 def adjust_longitude(obs_DF, osw_source):
-    if osw_source in ('CYGNSS', 'Spire'):
+    if osw_source in ('CYGNSS', 'SPIRE', 'Spire'):
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
    if osw_source in ('CYGNSS', 'Spire', 'Spire-L2'):
```

File: src/hdf5/osw_2ioda.py [hunk 5]
```diff
@@ -309,11 +314,13 @@ def add_long_description(osw_source):
...
     if osw_source == 'CYGNSS':
-        long_description = 'Derived from CYGNSS GNSS-R recievers.'
+        long_description = 'OSW derived from CYGNSS GNSS-R recievers.'
     elif osw_source == 'Muon':
-        long_description = 'Proxy Muon data derived from CYGNSS GNSS-R recievers.'
+        long_description = 'Proxy Muon OSW data derived from CYGNSS GNSS-R recievers.'
     elif osw_source == 'Spire':
-        long_description = 'Derived from Spire GNSS-R recievers.'
+        long_description = 'OSW derived from Spire GNSS-R recievers.'
+    elif osw_source == 'SPIRE':
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
    elif osw_source == 'Spire-L2':
```

File: src/hdf5/osw_2ioda.py [hunk 6]
```diff
@@ -309,11 +314,13 @@ def add_long_description(osw_source):
     # Add some additional information to platformLongDescription global attribute
     long_description = ''
     if osw_source == 'CYGNSS':
-        long_description = 'Derived from CYGNSS GNSS-R recievers.'
+        long_description = 'OSW derived from CYGNSS GNSS-R recievers.'
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
        long_description = 'OSW derived from CYGNSS GNSS-R receivers.'
```

File: src/hdf5/osw_2ioda.py [hunk 7]
```diff
@@ -309,11 +314,13 @@ def add_long_description(osw_source):
...
     elif osw_source == 'Muon':
-        long_description = 'Proxy Muon data derived from CYGNSS GNSS-R recievers.'
+        long_description = 'Proxy Muon OSW data derived from CYGNSS GNSS-R recievers.'
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
        long_description = 'Proxy Muon OSW data derived from CYGNSS GNSS-R receivers.'
```

File: src/hdf5/osw_2ioda.py [hunk 8]
```diff
@@ -309,11 +314,13 @@ def add_long_description(osw_source):
...
     elif osw_source == 'Spire':
-        long_description = 'Derived from Spire GNSS-R recievers.'
+        long_description = 'OSW derived from Spire GNSS-R recievers.'
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
        long_description = 'OSW derived from Spire GNSS-R receivers.'
```

File: src/hdf5/osw_2ioda.py [hunk 9]
```diff
@@ -309,11 +314,13 @@ def add_long_description(osw_source):
...
+    elif osw_source == 'SPIRE':
+        long_description = 'NOAA Level-2 OSW from Spire GNSS-R recievers.'
```
Comment (fabiolrdiniz) [inline src/hdf5/osw_2ioda.py]:
```suggestion
        long_description = 'NOAA Level-2 OSW from Spire GNSS-R receivers.'
```

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @BenjaminRuston! What do you think about `Spire-L2` instead of `SPIRE`?

**Comment by BenjaminRuston** [comment]
@smaticka can you have a look at this one, it's an update that is allowing a NOAA produced Level-2 product of the Spire OSW to be decoded as well

File: src/hdf5/osw_2ioda.py [hunk 10]
```diff
@@ -213,10 +213,15 @@ def get_data_source(afile):
...
+    elif 'title' in afile.attrs.keys() and 'SPIRE' in afile.attrs['title'].decode('UTF-8'):
+        return 'Spire-L2'
+    else:
+        # throw error
+        pass
 
 
 def get_reference_time(afile, osw_source):
-    if osw_source in ('CYGNSS'):
+    if osw_source in ('CYGNSS') or 'Spire-L2' in osw_source:
```
Comment (smaticka) [inline src/hdf5/osw_2ioda.py]: Should this be?:
```suggestion
    if osw_source == 'CYGNSS' or osw_source == 'Spire-L2':
```

File: src/hdf5/osw_2ioda.py [hunk 11]
```diff
@@ -298,7 +303,7 @@ def adjust_dateTime(obs_DF, dat_ref):
 
 
 def adjust_longitude(obs_DF, osw_source):
-    if osw_source in ('CYGNSS', 'Spire'):
+    if osw_source in ('CYGNSS', 'Spire', 'Spire-L2'):
         mask = obs_DF['longitude'] > 180
         obs_DF.loc[mask, 'longitude'] = obs_DF['longitude'].loc[mask].values - 360
```
Comment (BenjaminRuston) [inline src/hdf5/osw_2ioda.py]:
```suggestion
    obs_DF.loc[obs_DF['longitude'] > 180, 'longitude'] -= 360
```

Comment (BenjaminRuston) [reply src/hdf5/osw_2ioda.py]: and thinking the section doesn't even need a conditional check

basically if *any* longitude provided is greater than 180,, then subtract by 360

**Review by smaticka — APPROVED** [review]
looks good to me. thanks @BenjaminRuston 

the way the if statements are written with the `==` I believe requires it be exactly equal, so it won't work for substrings. in which case, we could use a tuple as well which will keep the code shorter if the list gets long.

**Comment by BenjaminRuston** [comment]
thanks @smaticka 

note that this:
```
osw_source in ('CYGNSS') 
```

is equivalent to this:

```
osw_source == 'CYGNSS'
```


to check for a substring this would work:

```
'CYGNSS' in osw_source
```

**Pull request merged**